4_ASP.NET_CORE/task4/server_task4/server_task4/Services/CommentsService.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using server_task4.DAL.Context;
+using server_task4.DAL.Models;
+using server_task4.Models;
+
+namespace server_task4.Services
+{
+    public class CommentsService : ICommentsService
+    {
+        private dbContext db;
+        private readonly IMapper _mapper;
+        private readonly IUserService _userService;
+        public CommentsService(dbContext comments, IMapper mapper, IUserService userService)
+        {
+            this._mapper = mapper;
+            this.db = comments;
+            this._userService = userService;
```

Comment:
You should not add dependency to UserService just because you need to resolve UserName